lib/streamlit/watcher/local_sources_watcher.py

```diff
@@ -192,7 +192,7 @@ def get_module_paths(module: ModuleType) -> set[str]:
         # __file__ is the pathname of the file from which the module was loaded
         # if it was loaded from a file.
         # The __file__ attribute may be missing for certain types of modules
-        lambda m: [m.__file__],
+        lambda m: [m.__file__] if hasattr(m, "__dict__") and "__file__" in m.__dict__ else [],
```

Comment:
Think it may be more comprehensive to simplify check - will then handle all cases where the attribute is accessible, regardless of whether it's stored in `__dict__` 
```python
lambda m: [m.__file__] if hasattr(m, "__file__") else []
```